### PR TITLE
Feature/fix urls - WIP

### PR DESCRIPTION
**Pull request by moradology**
A bit of a misnomer here - included is also some styling for John's demo of interest, perhaps, to @flibbertigibbet as well

### CODE REVIEW

**Comment by moradology** [comment]
Ah, yes. I usually use 'asdf' before I squash it - must have forgotten somehow.

File: web/app/scripts/stepwise/stepwise-directive.js [hunk 1]
```diff
@@ -75,7 +75,7 @@
 
                     svg.append('g')
                       .attr('class', 'x axis')
-                      .call(xAxis)
+                      //.call(xAxis)
```
Comment (kshepard) [inline web/app/scripts/stepwise/stepwise-directive.js]: Remove

File: web/app/scripts/views/dashboard/dashboard-partial.html [hunk 2]
```diff
@@ -1,13 +1,136 @@
...
-            <div class="form-area-heading">
-                <h2>Dashboard page placeholder</h2>
-            </div>
-            <div driver-stepwise chart-data="ctl.records.results"></div>
-            <div driver-toddow chart-data="ctl.records.results"></div>
-            <div class="form-area-body">
-            </div>
-        </div>
-    </div>
+<main class="dashboard">
```
Comment (kshepard) [inline web/app/scripts/views/dashboard/dashboard-partial.html]: This file can use some formatting.

Comment (moradology) [inline web/app/scripts/views/dashboard/dashboard-partial.html]: This HTML is purely for the demo on tues; i'm not sure how important the text you see here actually is for readability/reuse/maintenance

Comment (kshepard) [inline web/app/scripts/views/dashboard/dashboard-partial.html]: I imagine we'll probably end up using some of it afterwards though when fleshing out the dashboard. I wasn't thinking anything excessive, just having your editor auto-format the tags.

**Comment by kshepard** [comment]
I took this branch for a spin and there are a few things that aren't working correctly. Switching record types doesn't seem to do anything: for example, when you're on the list view and change the record type, the new set of records aren't pulled down. Also, nothing is being persisted on page refresh -- it's always choosing the default options in the navbar dropdowns. If you're on the list view and change the the record type and geometry and then refresh the page, it should maintain those selected values and also apply them to the page. Also, the dashboard styling looks pretty weird: there are a lot of squashed-together elements.

**Comment by moradology** [comment]
I'll be adding commits to this tomorrow addressing the tests and missing functionality. This should have been marked WIP.

**Comment by moradology** [comment]
Functionality is nicely in place and I'm currently working on testing - angular-local-storage introduces some new difficulties in testing that I think I'm going to have to use jasmine's spy for.

File: web/app/scripts/state/recordstate-service.js [hunk 3]
```diff
@@ -20,6 +27,8 @@
          */
         function init() {
           selected = null;
+          gettingSelected = false;
+          gettingOptions = false;
```
Comment (kshepard) [inline web/app/scripts/state/recordstate-service.js]: Just curious: why are these variables needed in this service, but not the other services?

File: web/app/scripts/stepwise/stepwise-directive.js [hunk 4]
```diff
@@ -10,21 +10,31 @@
...
-                var margin = {top: 20, right: 20, bottom: 30, left: 50},
-                    width = 752 - margin.left - margin.right,
-                    height = 475 - margin.top - margin.bottom;
+            link: function(scope, elem, attrs, elemStat) {
+                scope.$watchGroup([
+                    function() { return elemStat.height; },
+                    function() { return elemStat.width; }],
+                    function(oldVal, newVal) {
+                        console.log('old', oldVal, 'new', newVal);
+                        console.log(elemStat.height, elemStat.width);
```
Comment (kshepard) [inline web/app/scripts/stepwise/stepwise-directive.js]: These can either be removed, or if needed, switched to use $log.

File: web/app/scripts/stepwise/stepwise-directive.js [hunk 5]
```diff
@@ -52,52 +62,64 @@
...
                         .scale(y)
                         .orient('left')
-                        .ticks(Math.min(10, _.max(_.map(data, function(d) { return d.count; }))))
                         .tickSize(1);
 
                     svg.append('g')
-                      .attr('class', 'x axis')
-                      .call(xAxis)
+                      .attr('class', 'xAxis')
+                      //.call(xAxis)  // TODO: uncomment after worldbank demo
```
Comment (kshepard) [inline web/app/scripts/stepwise/stepwise-directive.js]: What happens when we eventually uncomment this? Does it break anything in particular?

File: web/app/scripts/stepwise/stepwise-directive.js [hunk 6]
```diff
@@ -52,52 +62,64 @@
...
+                 * @param data {array} the data to change our chart with
+                 */
+                function updateChart(data) {
+                    y.domain([0, d3.max(data, function(d) { return d.count; })]);
+                    yAxis.scale(y)
+                        .ticks(Math.min(10, _.max(_.map(data, function(d) { return d.count; }))));
+                    xAxis.scale(x)
+                        .tickFormat(xAxisTextFormat);
+
+                    svg.select('.xAxis').transition().call(xAxis);  // TODO: uncomment
```
Comment (kshepard) [inline web/app/scripts/stepwise/stepwise-directive.js]: Should this be commented, or the comment removed?

**Comment by kshepard** [comment]
This is looking good. The only weird thing I noticed is that clicking on the state dropdowns and not making a selection changes the url to the root state. It'd be good to fix that if possible. +1 after fixing linting/tests.

File: web/app/scripts/state/recordstate-service.js (hunk 3 above)
Comment (moradology) [inline web/app/scripts/state/recordstate-service.js]: They should definitely be extended to the other scripts - the main purpose is just to reduce the number of outgoing requests.

**Pull request merged**